data/basal.go

```diff
@@ -0,0 +1,166 @@
+package data
+
+import (
+	"fmt"
+	"reflect"
+	"strings"
+
+	valid "github.com/tidepool-org/platform/Godeps/_workspace/src/gopkg.in/bluesuncorp/validator.v8"
```

Comment:
I see the use of `validator` below in the `init()` function.  As I mention later, I think the raw use of that global variable (that is defined elsewhere) should probably be wrapper in a getter (eg. `GetPlatformValidator()`) so you can rename the internal variable something other than `validator`. Then you can use the `validator` package as it is.